hbase-oss/src/main/java/org/apache/hadoop/hbase/oss/sync/ZKTreeLockManager.java

```diff
@@ -121,9 +120,26 @@ private void waitForCuratorToConnect() {
   }
 
   @Override
-  public void close() throws IOException {
+  public void close() {
     if (curator != null) {
-      curator.close();
+      synchronized(this) {
+        curator.close();
+        lockCache.clear();
+        logCaller();
```

Comment:
Checked CuratorFrameworkImpl and am happy to see that multiple calls to close() are OK to make.